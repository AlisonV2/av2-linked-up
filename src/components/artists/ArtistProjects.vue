<template>
  <div class="row" v-if="projects.length">
    <div class="col-12 col-md-6 col-xl-4" v-for="project in projects" :key="project">
      <router-link :to="{ name: 'ProjectDetails', params: { id: project.id, isArtist: true } }">
        <ProjectItem :project="project"/>
      </router-link>
    </div>
  </div>
</template>

<script>
import ProjectItem from '@/components/admin/ProjectItem';
import * as Sentry from '@sentry/vue';

/**
 * @exports ArtistProjects
 * @type {Component}
 * @requires Sentry
 * @vue-data {array} projects
 */
export default {
  name: 'ArtistProjects',
  components: {
    ProjectItem,
  },
  data() {
    return {
      projects: [],
    };
  },
  created() {
    this.$store
      .dispatch('getArtistProjects')
      .then(() => {
        this.projects = this.$store.getters.getArtistProjects;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>
