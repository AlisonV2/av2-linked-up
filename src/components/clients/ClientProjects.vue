<template>
  <div class="row"></div>
</template>

<script>
// import * as Sentry from '@sentry/vue';

/**
 * @exports ClientProjects
 * @type {Component}
 * @requires Sentry
 */
export default {
  name: 'ClientProjects',
  data() {
    return {};
  },
  methods: {},
};
</script>

<style></style>
