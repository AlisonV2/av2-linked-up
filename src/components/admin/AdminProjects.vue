<template>
  <ArtistProjects v-if="role === 'artist'"/>
  <ClientProjects v-if="role === 'client'"/>
</template>

<script>
import ArtistProjects from '@/components/artists/ArtistProjects.vue';
import ClientProjects from '@/components/clients/ClientProjects.vue';

/**
 * @exports AdminProjects
 * @type {Component}
 * @vue-data{string} role
 * @vue-event {number} getUserRole on created hook
 */
export default {
  name: 'AdminProfile',
  components: {
    ArtistProjects,
    ClientProjects,
  },
  data() {
    return {
      role: '',
    };
  },
  async created() {
    await this.$store.dispatch('getUserRole').then(() => {
      this.role = this.$store.getters.getUserRole;
    });
  },
};
</script>
