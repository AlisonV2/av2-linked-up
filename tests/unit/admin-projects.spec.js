import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import AdminProjects from '@/components/admin/AdminProjects.vue';
import ArtistProjects from '@/components/artists/ArtistProjects.vue';
import ClientProjects from '@/components/clients/ClientProjects.vue';

const actions = {
  getUserRole: jest.fn(),
};

const getters = {
  getUserRole: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const config = {
  data() {
    return {
      role: 'artist',
    };
  },
  global: {
    plugins: [store],
    components: {
      ArtistProjects,
      ClientProjects,
    },
  },
};

/**
 * @module AdminProjectsTest
 */
describe('AdminProjects.vue', () => {
  it('AdminProjects Snapshot', () => {
    const component = shallowMount(AdminProjects, config);
    expect(component.element).toMatchSnapshot();
  });

  it('getUserRole action should have been called', async () => {
    const component = shallowMount(AdminProjects, config);
    AdminProjects.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getUserRole).toHaveBeenCalled;
  });

  it('getUserRole getter should have been called', async () => {
    const component = shallowMount(AdminProjects, config);
    AdminProjects.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getUserRole).toHaveBeenCalled;
  });

  it('ArtistProjects should exist', () => {
    const component = shallowMount(AdminProjects, config);
    const item = component.findComponent(ArtistProjects);
    expect(item.exists()).toBe(true);
  });

  it('ClientProjects should exist', () => {
    const component = shallowMount(AdminProjects, {
      ...config,
      data() {
        return {
          role: 'client',
        };
      },
    });
    const item = component.findComponent(ClientProjects);
    expect(item.exists()).toBe(true);
  });
});
