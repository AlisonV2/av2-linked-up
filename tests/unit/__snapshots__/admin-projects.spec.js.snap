// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminProjects.vue AdminProjects Snapshot 1`] = `
<div
  data-v-app=""
>
  
  <artist-projects-stub />
  <!--v-if-->
  
</div>
`;
